src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -102,6 +104,14 @@ public void Dispose()
             this.meter?.Dispose();
         }
 
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionCounts()
+        {
+            for (int i = 0; i < NumberOfGenerations; ++i)
+            {
+                yield return new(GC.CollectionCount(i), new KeyValuePair<string, object>("gen", HeapNames[i]));
```

Comment:
The naming is a bit weird to me.
I suggest we keep them consistent, e.g. either use "gen" and "GenNames"; or "heap" and "HeapNames" (or "values") .

Reply:
It is indeed weird. I don't have a position on it since I'm confused about what to name it myself AND because it's a private variable. Using the words in a blog about [Internals of the POH](https://devblogs.microsoft.com/dotnet/internals-of-the-poh/) in .NET Blog:

> It's a bit unfortunate that we overloaded the word "heap" here but it's been this way since V1.0. When I started it was already too late to make changing the name worthwhile. But fortunately AFAICS it didn't seem that confusing for our customers.

https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals

According to article [Fundamentals of garbage collection](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals) plus the blog above, there are 3 generations of garbage collection and 5 portions of the managed heap. I've listed them here.

| Name for the garbage collection generation | Name for a portion of the managed heap                             |
|--------------------------------------------|--------------------------------------------------------------------|
| Generation 0                               | Generation 0                                                       |
| Generation 1                               | Generation 1                                                       |
| Generation 2                               | Generation 2                                                       |
|                                            | Large object heap (which is sometimes referred to as generation 3) |
|                                            | Pinned object heap                                                 |

Speaking of which I've realized that I never saw that Pinned Object Heap is referred to as generation 4, nor anyone mentioned `GC.GetGCMemoryInfo().GenerationInfo[4].SizeAfterBytes` describes Pinned Object Heap. I'll need to confirm on that too.

Reply:
I've renamed it to "GenNames" just to keep them seem consistent.